compiler/rustc_typeck/src/errors.rs

```diff
@@ -248,3 +251,62 @@ pub struct ExplicitGenericArgsWithImplTrait {
     #[help]
     pub is_nightly_build: Option<()>,
 }
+
+pub struct MissingTypeParams {
+    pub span: Span,
+    pub def_span: Span,
+    pub missing_type_params: Vec<String>,
+    pub empty_generic_args: bool,
+}
+
+impl<'a> SessionDiagnostic<'a> for MissingTypeParams {
```

Comment:
I think we should document for all manual impls why we didn't make them derives. Not in extreme detail, just a note about whatever made one use a manual impl

Reply:
I think that makes sense, I've added a comment here.